src/masternodes/mn_checks.cpp

```diff
@@ -3975,130 +3976,102 @@ Res HasAuth(const CTransaction &tx, const CCoinsViewCache &coins, const CScript
     return DeFiErrors::InvalidAuth();
 }
 
-enum class DomainTransferType : uint8_t {
-    DVM,
-    EVM,
-};
+static Res ValidateTransferDomainScripts(const CScript &srcScript, const CScript &destScript, VMDomainEdge aspect) {
+    CTxDestination src, dest;
+    auto res = ExtractDestination(srcScript, src);
+    if (!res) return DeFiErrors::ScriptUnexpected(srcScript);
 
-static bool DomainTransferAllowedAddress(const CScript &address, DomainTransferType type) {
```

Comment:
There are many places in mn_checks are further across the codebase use this exact check for PKHashType and WitV0KeyHashType. This function would have been used for them at a later date. I did not update them when adding this function to avoid further conflicts with the mn_check split PR.

Reply:
Yes, please. Let's further abstract from this point and lift up as needed to reuse if that ok to you?